Examples/OneSignalDemo/app/src/main/java/com/onesignal/sdktest/model/MainActivityViewModel.java

```diff
@@ -468,123 +460,97 @@ public void setupSettingsLayout() {
 
     private void setupSubscriptionSwitch() {
         boolean isPermissionEnabled = OneSignal
-                .getPermissionSubscriptionState()
-                .getPermissionStatus()
-                .getEnabled();
+                .getPermissionSubscriptionState() != null &&
+                OneSignal
+                        .getPermissionSubscriptionState()
+                        .getPermissionStatus()
+                        .getEnabled();
 
         final boolean isSubscribed = OneSignal
-                .getPermissionSubscriptionState()
-                .getSubscriptionStatus()
-                .getSubscribed();
+                .getPermissionSubscriptionState() != null &&
+                OneSignal
+                        .getPermissionSubscriptionState()
+                        .getSubscriptionStatus()
+                        .getSubscribed();
 
         subscriptionSwitch.setEnabled(isPermissionEnabled);
         subscriptionSwitch.setChecked(isSubscribed);
 
         if (isPermissionEnabled) {
-            subscriptionRelativeLayout.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(View v) {
-                    boolean isSubscribed = subscriptionSwitch.isChecked();
-                    subscriptionSwitch.setChecked(!isSubscribed);
-                    OneSignal.setSubscription(!isSubscribed);
-                }
+            subscriptionRelativeLayout.setOnClickListener(v -> {
+                boolean isSubscribed1 = subscriptionSwitch.isChecked();
+                subscriptionSwitch.setChecked(!isSubscribed1);
+                OneSignal.setSubscription(!isSubscribed1);
             });
         } else {
-            subscriptionRelativeLayout.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(View v) {
-                    intentTo.notificationPermissions();
-                }
-            });
+            subscriptionRelativeLayout.setOnClickListener(v -> intentTo.notificationPermissions());
         }
 
-        subscriptionSwitch.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                boolean isSubscribed = subscriptionSwitch.isChecked();
-                OneSignal.setSubscription(!isSubscribed);
-            }
+        subscriptionSwitch.setOnClickListener(v -> {
+            boolean isSubscribed12 = subscriptionSwitch.isChecked();
```

Comment:
Do lamdas not have their own scope so we can keep this `isSubscribed`?

Reply:
unfortunately, lambdas don't have custom scope :/